src/server/game/Instances/InstanceSaveMgr.cpp

```diff
@@ -516,10 +544,27 @@ void InstanceSaveManager::_ResetSave(InstanceSaveHashMap::iterator &itr)
     // do not allow UnbindInstance to automatically unload the InstanceSaves
     lock_instLists = true;
 
+    bool shouldDelete = true;
     InstanceSave::PlayerListType &pList = itr->second->m_playerList;
-    while (!pList.empty())
+    std::vector<Player*> temp; // list of expired binds that should be unbound
+    for (Player* player : pList)
     {
         Player* player = *(pList.begin());
+        if (InstancePlayerBind* bind = player->GetBoundInstance(itr->second->GetMapId(), itr->second->GetDifficulty()))
+        {
+            assert(bind->save == itr->second);
```

Comment:
Please use the TC assert macro to ensure the check is valid in every build or remove it if it's not that much needed. Having different checks with different build modes makes crash logs more confusing.